kong/conf_loader.lua

```diff
@@ -187,6 +187,28 @@ local CONF_INFERENCES = {
 
   lua_ssl_verify_depth = { typ = "number" },
   lua_socket_pool_size = { typ = "number" },
+
+  -- known keys which require no type inference:
+  prefix = {},
+  pg_user = {},
+  pg_host = {},
+  pg_schema = {},
+  pg_database = {},
+  cassandra_username = {},
+  cassandra_keyspace = {},
+  declarative_config = {},
+  mem_cache_size = {},
+  lua_package_path = {},
+  lua_package_cpath = {},
+  lua_ssl_trusted_certificate = {},
+  ssl_cert = {},
+  ssl_cert_key = {},
+  ssl_ciphers = {},
+  ssl_cipher_suite = {},
+  admin_ssl_cert = {},
+  admin_ssl_cert_key = {},
+  client_ssl_cert = {},
+  client_ssl_cert_key = {},
```

Comment:
Wasn't this unnecessary? Since the list of configuration values in the default template is considered the source of truth containing all valid properties.